src/misc/lv_mem.h

```diff
@@ -235,6 +235,14 @@ LV_ATTRIBUTE_FAST_MEM void lv_memset_ff(void * dst, size_t len);
 /**********************
  *      MACROS
  **********************/
+/**
+ * Allocate v to a memory area of size s, check and zero the returned area in one operation
+ */
+#define LV_ZALLOC(v, s) do { void * p = lv_mem_alloc(s); LV_ASSERT_MALLOC(p); if (p != NULL) lv_memset_00(p, s); memcpy(&v, &p, sizeof(p)); } while(0)
```

Comment:
* zalloc is for alloc & zero and it's used in linux. The lib C's standard name is calloc but it take a counter-intuitive count and item size parameter.
* no it's correct. I'm using `memcpy` to copy the pointer and skip the type checking from the compiler. If I had written `v = p;`, it would have failed to compile on C++ code, since p is a `void*` but `v` is any pointer type. So using memcpy convert everything to void * and it works. I could have use a `typeof` macro here to cast, but it's GCC only in C99.
* I can't use a static inline function here since C99 does not allow overload/generic and v is of any pointer type. So it has to be a macro until we rewrite LVGL as C++ :stuck_out_tongue_winking_eye:

Reply:
Fair enough :slightly_smiling_face: